sdk/monitor/azquery/metrics_client.go

```diff
@@ -0,0 +1,204 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+// DO NOT EDIT.
+
+package azquery
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+// MetricsClient contains the methods for the Metrics group.
+// Don't use this type directly, use NewMetricsClient() instead.
+type MetricsClient struct {
+	pl runtime.Pipeline
+}
+
+// NewListMetricDefinitionsPager - Lists the metric definitions for the resource.
+// Generated from API version 2018-01-01
+// resourceURI - The identifier of the resource.
+// options - MetricsClientListMetricDefinitionsOptions contains the optional parameters for the MetricsClient.ListMetricDefinitions
+// method.
+func (client *MetricsClient) NewListMetricDefinitionsPager(resourceURI string, options *MetricsClientListMetricDefinitionsOptions) *runtime.Pager[MetricsClientListMetricDefinitionsResponse] {
+	return runtime.NewPager(runtime.PagingHandler[MetricsClientListMetricDefinitionsResponse]{
+		More: func(page MetricsClientListMetricDefinitionsResponse) bool {
```

Comment:
How much desire do we have to rename or shorten some of these? Just the fact that it has `Metric` in the name is probably enough and we could omit the "MetricsClient" part of the type altogether.

Reply:
Good point, added to todo list for next release